pkg/gitops/gitops.go

```
@@ -677,6 +683,75 @@ func CreateGitOpsCommit(gitOpsConfig *GitOpsConfig, appSlug string, appName stri
 	return gitOpsConfig.CommitURL(updatedHash.String()), nil
 }
 
+// GetGitopsKeypairs will return all gitops keypairs in a namespace.
+func GetGitopsKeypairs(namespace string) (map[string]KeyPair, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get cluster config")
+	}
+
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create kubernetes clientset")
+	}
+
+	secret, err := clientset.CoreV1().Secrets(namespace).Get(context.TODO(), "kotsadm-gitops", metav1.GetOptions{})
+	if kuberneteserrors.IsNotFound(err) {
+		return nil, nil
+	} else if err != nil {
+		return nil, errors.Wrap(err, "get kotsadm-gitops secret")
+	}
+
+	indexes := map[int64]struct{}{}
+
+	for key, _ := range secret.Data {
+		splitKey := strings.Split(key, ".")
+		if len(splitKey) != 3 {
+			continue
+		}
+
+		newIndex, err := strconv.ParseInt(splitKey[1], 10, 64)
+		if err != nil {
+			continue
+		}
+		indexes[newIndex] = struct{}{}
+	}
+
+	encryptionKeySecret, err := clientset.CoreV1().Secrets(namespace).Get(context.TODO(), "kotsadm-encryption", metav1.GetOptions{})
+	if kuberneteserrors.IsNotFound(err) {
```

i think this should fail if the encryption key is not found